### ISSUE TITLE
Dropdown issues prop type warning for tabIndex on nested ListBoxField component

### ISSUE THREAD

**Issue by austinbruch** [carbon-design-system/carbon#3421]
Rendering the Dropdown component emits a console warning about the `tabIndex` prop on a nested `ListBoxField` component.

## What package(s) are you using?

- [X] `carbon-components`
- [X] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.

Rendering the Dropdown component emits a console warning about the `tabIndex` prop on a nested `ListBoxField` component.

> Is this issue related to a specific component?

`Dropdown`

> What did you expect to happen? What happened instead? What would you like to
> see changed?

No prop type warning that is outside my control.

> What browser are you working in?

Chrome

> What version of the Carbon Design System are you using?

carbon-components 10.4.1
carbon-components-react 7.4.1

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

## Steps to reproduce the issue

Render a `Dropdown` component.

> Please create a reduced test case in CodeSandbox
>
> - Style and vanilla JS:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/components/examples/codesandbox
> - React:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox

https://codesandbox.io/s/codesandbox-shklr

## Additional information

Console warning:
```
Warning: Failed prop type: Invalid prop `tabIndex` of value `0` supplied to `ListBoxField`, expected one of [null,null].
    in ListBoxField (created by Downshift)

    in Downshift (created by Dropdown)

    in div (created by Dropdown)

    in Dropdown (created by App)

    in div (created by App)

    in App
```

**Comment by jendowns** [comment]
Hey @austinbruch just a heads up -- I addressed this in PR #3425 but it wasn't automatically closed out because I didn't reference this issue number correctly in my desc 😇

**Comment by austinbruch** [comment]
Thanks, Jen.
